File: ios/CameraApp/Services/RustBridge.swift
/**
 * Rust shared library bridge layer
 * UniFFI auto-generates Swift bindings to the Generated/ directory.
 *
 * Build flow:
 * 1. cd shared-core && cargo build --release --target aarch64-apple-ios
 * 2. UniFFI generates Swift files to Generated/
 * 3. Xcode project references the Generated/ Swift files
 *
 * When bindings are generated, replace the placeholder implementations below
 * with actual camera_shared_core calls:
 *   - camera_shared_core.ImageProcessor().autoEnhance(inputImage:)
 *   - camera_shared_core.ImageProcessor().applyFilter(inputImage:filter:)
 *   - camera_shared_core.ImageProcessor().supportedFilters()
 *   - camera_shared_core.ImageProcessor().filterName(filter:)
 *   - camera_shared_core.getVersion()
 */

import Foundation

enum RustBridge {
    /// Check if Rust library is available
    static func isLoaded() -> Bool {
        // TODO: Replace with actual check when UniFFI bindings are generated
        return false
    }

    /// Get Rust library version
    static func getVersion() -> String {
        // TODO: Replace with: try camera_shared_core.getVersion()
        return "0.1.0 (bindings not generated)"
    }

    /// Get supported filter types
    static func supportedFilters() -> [FilterType] {
        // TODO: Replace with: ImageProcessor().supportedFilters().map { ... }
        return FilterType.allCases
    }

    /// Get display name for a filter
    static func filterName(_ filter: FilterType) -> String {
        // TODO: Replace with: ImageProcessor().filterName(filter: filter)
        return filter.rawValue
    }

    /// Auto-enhance image data
    static func autoEnhance(imageData: Data) async -> Data? {
        // TODO: Replace with:
        // let processor = camera_shared_core.ImageProcessor()
        // let result = try processor.autoEnhance(inputImage: Array(imageData))
        // return Data(result)
        return imageData
    }

    /// Apply filter to image data
    static func applyFilter(imageData: Data, filter: FilterType) async -> Data? {
        // TODO: Replace with:
        // let processor = camera_shared_core.ImageProcessor()
        // let filterType = mapToRustFilter(filter)
        // let result = try processor.applyFilter(inputImage: Array(imageData), filter: filterType)
        // return Data(result)
        return imageData
    }
}

File: android/app/src/main/java/com/camera/app/bridge/RustBridge.kt
package com.camera.app.bridge

/**
 * Rust 共享库桥接层
 * UniFFI 会自动生成绑定代码，这里做初始化和加载
 *
 * 编译流程：
 * 1. cd shared-core && cargo build --release --target aarch64-linux-android
 * 2. 把 libcamera_shared_core.so 放到 android/app/src/main/jniLibs/arm64-v8a/
 * 3. UniFFI 生成的 Kotlin 类会自动加载这个 .so
 */
object RustBridge {

    init {
        try {
            System.loadLibrary("camera_shared_core")
        } catch (e: UnsatisfiedLinkError) {
            // 开发阶段 .so 可能还未编译，先跳过
            println("⚠️ Rust shared library not loaded: ${e.message}")
        }
    }

    /**
     * 检查 Rust 库是否已加载
     */
    fun isLoaded(): Boolean {
        return try {
            // 调用 UniFFI 生成的函数来验证
            uniffi.camera_shared_core.get_version()
            true
        } catch (e: Throwable) {
            false
        }
    }

    /**
     * 获取 Rust 库版本
     */
    fun getVersion(): String {
        return try {
            uniffi.camera_shared_core.get_version()
        } catch (e: Throwable) {
            "not loaded"
        }
    }
}